### DISCUSSION TITLE
Has anyone successfully turned on iommu on rk3588?

### DISCUSSION THREAD

**Discussion by wanlce** [Joshua-Riek/ubuntu-rockchip#354] [Q&A]
I apologize in advance, my device is nanopc-t6 rk3588, using pve installed on armbian, I can't use pci passthrough, I would like to ask if anyone has succeeded in pci passthrough!
![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/55907021/1f72fda9-bff7-4e1c-9ffb-e358e1517cc0)

![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/55907021/26968bad-829a-4259-bc1f-d461f59bf183)
![image](https://github.com/Joshua-Riek/ubuntu-rockchip/assets/55907021/b65e4928-ab3b-4a44-a9a6-f74a341db168)

**Comment by SuperKali** [comment]
@wanlce how you've installed it?

**Reply by wanlce** [reply]
armbian  https://github.com/armbian/build
Proxmox https://github.com/jiangcuo/Proxmox-Arm64

**Reply by SuperKali** [reply]
Thank you, i will try it with a SD card.

**Reply by SuperKali** [reply]
@wanlce i have the same problem no pcie passthrough

**Comment by insomnia417** [comment]
+1

**Comment by brightplastik** [comment]
I would greet any improvement in the IOMMU situation, as my main aim with this board is to develop a proxmox server. ATM with broken IOMMU it is quite a PITA.

**Comment by RadxaStephen** [comment]
> I apologize in advance, my device is nanopc-t6 rk3588, using pve installed on armbian, I can't use pci passthrough, I would like to ask if anyone has succeeded in pci passthrough! ![image](https://private-user-images.githubusercontent.com/55907021/266615859-1f72fda9-bff7-4e1c-9ffb-e358e1517cc0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyODI1NTMsIm5iZiI6MTcxODI4MjI1MywicGF0aCI6Ii81NTkwNzAyMS8yNjY2MTU4NTktMWY3MmZkYTktYmZmNy00ZTFjLTlmZmItZTM1OGUxNTE3Y2MwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEzVDEyMzczM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5ZGUxZjQ0ZTE4ZWU3N2Y5MDVlMzUyMjQ4Nzk1MzQ3MDhmZmE5ZWQzZDMwYzIwZWIxODQ4Yzc3NGVkZTZkMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DVVtNKxlMJDyu-wrsnEFyJqCuugtOs-y3X05gQgusGk)
> 
> ![image](https://private-user-images.githubusercontent.com/55907021/266615375-26968bad-829a-4259-bc1f-d461f59bf183.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyODI1NTMsIm5iZiI6MTcxODI4MjI1MywicGF0aCI6Ii81NTkwNzAyMS8yNjY2MTUzNzUtMjY5NjhiYWQtODI5YS00MjU5LWJjMWYtZDQ2MWY1OWJmMTgzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEzVDEyMzczM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkZWFlMTY1ZGFiNzg3MDRhYTQwMzUxZDNlMzc5OTlkODdmZGU4NDhmN2FkYTUwNGZmNzEzOWRlMjZiNzg2ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.u6FhgVnqfOwmZI7aPbmjFRRfDtMMuV_DvfxqJYix7Xk) ![image](https://private-user-images.githubusercontent.com/55907021/266615567-b65e4928-ab3b-4a44-a9a6-f74a341db168.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyODI1NTMsIm5iZiI6MTcxODI4MjI1MywicGF0aCI6Ii81NTkwNzAyMS8yNjY2MTU1NjctYjY1ZTQ5MjgtYWIzYi00YTQ0LWE5YTYtZjc0YTM0MWRiMTY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEzVDEyMzczM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlZDg5NjFjNDU2OTE3MDBkMjQ2MzVlNTdkNWM4ZGU0NGVhNGJiZTFmYzUwNzdlNDczNjRlNjJkNTM1ODhmMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1-mFmXr2YC3dsLjcoo5QyKC8kOikZEAYOoNzEUmRn-A)

For RK3588 PCIe pass through, this guide may help you. [RK3588 PCIe Virtualization](https://dl.radxa.com/users/dev/Rockchip_PCIe_Virtualization_Developer_Guide_CN.pdf)

**Reply by brightplastik** [reply]
Fantastic Stephen! It's the first evidence something is moving, since months! Thank you for this excellent tip!

**Reply by scott87** [reply]
Did you ever figure this out. Been messing around for a while and not getting anywhere. Any help would be much appreciated.

**Reply by brightplastik** [reply]
> Did you ever figure this out. Been messing around for a while and not getting anywhere. Any help would be much appreciated.

Hello Scott, no unfortunately I didn't even start implementing what Stephen suggested.
But in the meantime something is moving, I think.
https://patchwork.kernel.org/project/linux-rockchip/patch/20241107123732.1160063-2-cassel@kernel.org/
At this link I see on 2nd December someone is applying a patch to show the pci segment of rk3588. Dunno what it means for us in practice. Surely @Joshua-Riek knows better where to start in order to adopt this future kernel improvement.
